### DISCUSSION TITLE
Updating privatebin from 1.4.0

### DISCUSSION THREAD

**Discussion by janstieler** [PrivateBin/PrivateBin#1127] [Q&A]
Hi,
is there a way to update privatebin from 1.4.0 without loosing my configuration and changes?
There is no update manual included.

It looks like the software is a little bit oldstylisch programmed.

**Comment by r4sas** [comment]
Upgrade manual is exists in project's wiki: https://github.com/PrivateBin/PrivateBin/wiki/Upgrading-from-ZeroBin-0.19-Alpha#upgrade-process. Even though it was written to upgrade from ZeroBin, nothing has changed.

**Comment by elrido** [comment]
In a nutshell, keep the data and cfg folder, replace the rest, i.e. copy the application over your existing install.

If you are into more modern paradigms, consider using one of the [container images](https://github.com/PrivateBin/docker-nginx-fpm-alpine#privatebin-on-nginx-php-fpm--alpine), either by itself or via [Helm](https://github.com/PrivateBin/helm-chart#privatebin-helm-chart), if you use Kubernetes. Then the upgrade consists of just replacing one version of the image with another, keeping the data and optional config volumes preserved outside the image.
